addons/payment_xendit/models/payment_transaction.py

```diff
@@ -140,11 +152,48 @@ def _process_notification_data(self, notification_data):
         if payment_status in const.PAYMENT_STATUS_MAPPING['pending']:
             self._set_pending()
         elif payment_status in const.PAYMENT_STATUS_MAPPING['done']:
+            if self.tokenize and not self.token_id:
```

Comment:
Why do you need to check if a token is not already created? This situation should be impossible as we write the token and `tokenize` at the same time on the transaction. 
```suggestion
            if self.tokenize:
```